miso-service/src/main/java/uk/ac/bbsrc/tgac/miso/service/BarcodableViewService.java

```diff
@@ -0,0 +1,16 @@
+package uk.ac.bbsrc.tgac.miso.service;
+
+import uk.ac.bbsrc.tgac.miso.core.data.Barcodable;
+import uk.ac.bbsrc.tgac.miso.core.data.impl.view.BarcodableView;
+
+import java.io.IOException;
+import java.util.Collection;
+import java.util.List;
+
+public interface BarcodableViewService {
+  List<BarcodableView> searchByBarcode(String barcode);
+
+  List<BarcodableView> searchByBarcode(String barcode, Collection<Barcodable.EntityType> typeFilter);
+
+  <T extends Barcodable> T getEntity(BarcodableView view) throws IOException;
```

Comment:
Please add Javadoc, something along the lines of "Fetch the full entity object represented by the provided BarcodableView. Return type depends on view.id.targetType. Incorrect usage (e.g. `view.id.targetType != SAMPLE && Sample s = getEntity(view)`) will result in ClassCastException"